shared/ui-components/src/alerts/SiteAlert.tsx

```diff
@@ -1,36 +1,37 @@
 import React, { useEffect, useState } from "react"
-import { useRouter } from "next/router"
 import { AlertBox } from "./AlertBox"
 import { AlertTypes } from "./alertTypes"
 
-type UrlAlertProps = {
-  urlParam?: string
+type SiteAlertProps = {
   timeout?: number
   dismissable?: boolean
   type?: AlertTypes
   className?: string
 }
 
+export const setSiteAlertMessage = (message: string, type: AlertTypes) => {
+  sessionStorage.setItem(`alert_message_${type}`, message)
+}
+
 /**
  * Show an alert based on a url query param.
  */
-export const UrlAlert = ({
-  urlParam = "message",
+export const SiteAlert = ({
   timeout,
   dismissable = true,
   type = "alert",
   className,
-}: UrlAlertProps) => {
+}: SiteAlertProps) => {
   const [open, setOpen] = useState(false)
-
-  const router = useRouter()
-  const {
-    query: { [urlParam]: message },
-  } = router
+  const [message, setMessage] = useState("")
 
   useEffect(() => {
     let timeoutRef: number
-    if (message) {
+
+    setMessage(sessionStorage.getItem(`alert_message_${type}`) || "")
+
+    if (message != "") {
```

Comment:
This effect is going to unnecessarily run at least twice and cause the component to render 3 times when we have a message:

1) triggered on first render, the `setMessage` call queues up a state update to message. However, `message` is still `""` on that render, so the rest of the logic doesn't fire.

2) The component re-renders in response to the state update. This time, `message` has changed based on the `setMessage` call in the previous render, and since it is in the effect dependency list, we got through the function again, including the `setMessage` call (fortunately, strings are immutable primitives in JS and React is smart enough to not blindly re-render, so the second call to `setMessage` with the same string doesn't trigger an infinite loop - if you were setting an object rather than a primitive this pattern would trigger an infinite re-render loop). Then we remove the item from `sessionStorage`, and `setOpen`, triggering a second re-render.

You can re-write this to avoid the unnecessary re-render:

```typescript
useEffect(() => {
  let timeoutRef: number
  const storedMessage = sessionStorage.getItem(...)
  if (storedMessage) {
    setMessage(storedMessage)
    setOpen(true)
    sessionStorage.removeItem()

    // timeout logic
  }
  return () => clearTimeout(timeoutRef)
}, [timeout])
```

Also, small note: use `!==`. To a first approximation, you never want to use `==` in JS - it performs type conversions automatically and way too aggressively, and has generally been the source of many a difficult-to-understand bug (not that I think it's causing an issue _here_ per se, but it's a good habit to get into writing JS code)

Reply:
Sorry re: `!=`…old Ruby habits die hard! At any rate, I reworked it with your suggestion and that makes sense. Sounds like this is now RTM

Reply:
> Sorry re: !=…old Ruby habits die hard! At any rate, I reworked it with your suggestion and that makes sense. Sounds like this is now RTM

@jaredcwhite Not just Ruby! Pretty much ANY sensible programming language other than JS...